src/node_buffer.cc

```diff
@@ -364,7 +370,11 @@ MaybeLocal<Object> New(Isolate* isolate,
                        void* hint) {
   EscapableHandleScope handle_scope(isolate);
   Environment* env = Environment::GetCurrent(isolate);
-  CHECK_NOT_NULL(env);  // TODO(addaleax): Handle nullptr here.
+  if (env == nullptr) {
+    callback(data, hint);
```

Comment:
Belated review: this `callback()` call and the `free()` call on line 417 look a bit incongruent because they're not executed when the `Buffer::New()` calls below them fail.

Performing the action only sometimes is confusing, IMO. A caller that diligently checks the return value and frees the memory when it's empty will now (sometimes) double-free.

Reply:
@bnoordhuis I think it was more or less the idea to be more consistent here and always take ownership, because in other cases we already did free the data when failing (e.g. in the `New(Isolate*, Local<String>, encoding)` variant)…